oar/core/advisory.py

```diff
@@ -629,3 +662,103 @@ def has_blocking_secruity_alert(self):
             return blocking
         else:
             return False
+        
+    def get_unhealthy_builds(self):
+        """
+        Get unhealthy builds of advisory.
+
+        Returns:
+            list: Unhealthy builds.
+        """
+        nvrs = self.get_build_nvrs()
+
+        unhealthy_builds = []
+
+        for n in nvrs:
+            build_grades = self.get_build_grades(n)
+            for bg in build_grades:
+                if not util.is_grade_healthy(bg["grade"]):
+                    unhealthy_builds.append(bg)
+
+        return unhealthy_builds
+        
+    def get_build_grades(self, nvr):
+        """
+        Get all architecture grades of build.
+
+        Returns:
+            list: All architecture grades of build {nvr, architecture, grade}.
+        """
+        nvr_url = "https://pyxis.engineering.redhat.com/v1/images/nvr/%s?include=data.freshness_grades&include=data.architecture" % nvr
+        resp = requests.get(nvr_url, auth=self._auth, verify=self.ssl_verify)
+
+        nvr_grades = []
+
+        if resp.ok:
+            # TODO should we skip empty data? 
```

Comment:
@rioliu-rh If there is no data for nvr (no grade, etc), should we just skip the nvr?

Reply:
we should skip grade info check for rhcos builds like
```
Build rhcos-aarch64-414.92.202411260513-0
Build rhcos-ppc64le-414.92.202411260513-0
Build rhcos-s390x-414.92.202411260513-0
Build rhcos-x86_64-414.92.202411260513-0
```
skip this check for the nvr startwith `rhcos-$arch-`

Reply:
Those builds are automatically skipped because they don't have any `data`. See https://pyxis.engineering.redhat.com/v1/images/nvr/rhcos-s390x-414.92.202410290222-0

and the code is
```python
for arch in resp.json()["data"]:
```

Reply:
that's all right